### PR TITLE
Recover from Android 4.2.2 EBADF crashes.

### PR DESCRIPTION
**Pull request by swankjesse**
Originally submitted by @pguilbot.

See also https://github.com/square/okhttp/issues/1684

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/internal/Util.java [hunk 1]
```diff
@@ -277,4 +279,13 @@ public static String toHumanReadableAscii(String s) {
     }
     return s;
   }
+
+  /**
+   * Returns true if {@code e} is due to a firmware bug fixed after Android 4.2.2.
+   * https://code.google.com/p/android/issues/detail?id=54072
+   */
+  public static boolean isAndroidGetsocknameError(AssertionError e) {
```
Comment (JakeWharton) [inline okhttp/src/main/java/com/squareup/okhttp/internal/Util.java]: This method name makes my left eye twitch.

**Comment by JakeWharton** [comment]
LGTM

File: okhttp/src/main/java/com/squareup/okhttp/internal/Util.java (hunk 1 above)
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/Util.java]: It's deliberately ugly, named after the ugliness it bandaids.

**Comment by swankjesse** [comment]
@pguilbot I'm doing a round of releases tonight.

**Pull request merged**